crates/rust-eigenda-v2-client/src/payload_disperser.rs

```diff
@@ -103,26 +119,127 @@ impl<S> PayloadDisperser<S> {
             .disperser_client
             .blob_status(blob_key)
             .await
-            .map_err(|e| EigenClientError::PayloadDisperser(PayloadDisperserError::Disperser(e)))?;
+            .map_err(|e| {
+                EigenClientError::PayloadDisperser(Box::new(PayloadDisperserError::Disperser(e)))
+            })?;
 
-        let blob_status = BlobStatus::try_from(status.status)
-            .map_err(|e| EigenClientError::PayloadDisperser(PayloadDisperserError::Decode(e)))?;
+        let blob_status = BlobStatus::try_from(status.status).map_err(|e| {
+            EigenClientError::PayloadDisperser(Box::new(PayloadDisperserError::Decode(e)))
+        })?;
         match blob_status {
             BlobStatus::Unknown | BlobStatus::Failed => Err(PayloadDisperserError::BlobStatus)?,
-            BlobStatus::Encoded | BlobStatus::GatheringSignatures | BlobStatus::Queued => Ok(None),
+            BlobStatus::Encoded | BlobStatus::Queued => Ok(None),
+            BlobStatus::GatheringSignatures => {
+                let thresholds_met = self.check_thresholds(&status).await;
+                if thresholds_met.is_err() {
+                    // Since we are gathering signatures, it is ok for thresholds not to be met yet.
+                    return Ok(None);
+                }
+                let eigenda_cert = self.build_eigenda_cert(&status).await?;
+                self.cert_verifier
+                    .check_da_cert(&eigenda_cert)
+                    .await
+                    .map_err(|e| {
+                        EigenClientError::PayloadDisperser(Box::new(
+                            PayloadDisperserError::CertVerifier(e),
+                        ))
+                    })?;
+                Ok(Some(eigenda_cert))
```

Comment:
Can you update the function documentation with this new info (that cert is checked and an error is returned if cert is invalid).
Also please remove notion of `inclusion data`, that is zksync specific and shouldnt be part of the generic eigenda client.